MapboxVision/src/main/java/com/mapbox/vision/sync/visionpro/VisionProSyncManager.kt

```diff
@@ -0,0 +1,178 @@
+package com.mapbox.vision.sync.visionpro
+
+import com.google.gson.Gson
+import com.mapbox.vision.models.video.VideoMetadata
+import com.mapbox.vision.sync.SyncClient
+import com.mapbox.vision.sync.SyncManager
+import com.mapbox.vision.sync.filemanager.SyncFileHandler
+import com.mapbox.vision.sync.util.TotalBytesCounter
+import com.mapbox.vision.sync.util.VisionProEnvironment
+import com.mapbox.vision.utils.VisionLogger
+import com.mapbox.vision.utils.prefs.TotalBytesCounterPrefs
+import com.mapbox.vision.utils.threads.WorkThreadHandler
+import java.io.File
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.MediaType
+import okhttp3.MultipartBody
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+
+internal class VisionProSyncManager(
+    private val gson: Gson,
+    private val syncClient: SyncClient<OkHttpClient, VisionProEnvironment>,
+    private val syncQueue: VisionProQueue,
+    syncFileHandler: SyncFileHandler<VisionProEnvironment>
+) : SyncManager.SyncManagerBase<VisionProEnvironment>(syncQueue, syncFileHandler, MAX_DIR_SIZE) {
+
+    private val httpClient = syncClient.client
```

Comment:
maybe move _client_ and _byteCounter_ to SyncManager params? and use current implementation as default?
It'll be easier to test such classes